src/EnergyPlus/OutdoorAirUnit.cc

```diff
@@ -507,15 +507,17 @@ namespace OutdoorAirUnit {
             }
 
             // Process the unit control type
-
             if (!lAlphaBlanks(9)) {
-                {
-                    auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(9));
-                    if (SELECT_CASE_var == "NEUTRALCONTROL") {
-                        OutAirUnit(OAUnitNum).controlType = OAUnitCtrlType::Neutral;
-                    } else if (SELECT_CASE_var == "TEMPERATURECONTROL") {
-                        OutAirUnit(OAUnitNum).controlType = OAUnitCtrlType::Temperature;
-                    }
+                constexpr std::array<std::string_view, static_cast<int>(OAUnitCtrlType::Num)> ctrlTypeNamesUC = {
+                    "NEUTRALCONTROL", "INVALID-UNCONDITIONED", "TEMPERATURECONTROL"};
+                auto tmpCtrlType = static_cast<OAUnitCtrlType>(getEnumerationValue(ctrlTypeNamesUC, state.dataIPShortCut->cAlphaArgs(9)));
+                switch (tmpCtrlType) {
+                case OAUnitCtrlType::Neutral:
+                case OAUnitCtrlType::Temperature:
+                    OutAirUnit(OAUnitNum).controlType = tmpCtrlType;
+                    break;
+                default:
+                    break; // just leave it alone, nothing was done here
```

Comment:
This one is kinda funny, because the enumeration has 3 items, but we only have meaningful strings for 2 of them in this input object.  So one string is "invalid" in terms of this object, but could be valid for other objects which use that enum.  That screams that we are abusing that enum to be, but I think modifying this to use this new approach will make it easier to make a cleanup like that later.  Anyway, I kept the array of strings close by here so that no one else would attempt to use it.